<?xml version="1.0" encoding="utf-8" ?>
<launch>
    <let if="$(env IS_ROBOT false)" name="taskset" value="taskset -c 7"/>
    <let unless="$(env IS_ROBOT false)" name="taskset" value=""/>

    <!-- camera driver -->
    <node pkg="pylon_ros2_camera_wrapper" exec="pylon_ros2_camera_wrapper" name="pylon_camera_node" output="screen" launch-prefix="$(var taskset)">
        <param from="$(find-pkg-share bitbots_basler_camera)/config/camera_settings.yaml" />
        <param name="device_user_id" value="$(env ROBOT_NAME default)" />
        <param name="camera_info_url" value="package://bitbots_basler_camera/config/camera_calibration_$(env ROBOT_NAME default).yaml" />
        <param name="startup_user_set" value="CurrentSetting" />
        <param name="enable_status_publisher" value="true" />
        <param name="enable_current_params_publisher" value="true" />
    </node>
    <!-- Debayer + Binning -->
    <node pkg="bitbots_basler_camera" exec="postprocessing" name="basler_camera_post_processing" output="screen" launch-prefix="$(var taskset)">
        <remap from="in/image_raw" to="/pylon_camera_node/image_raw" />
        <remap from="in/camera_info" to="/pylon_camera_node/camera_info" />
        <remap from="out/image_proc" to="/camera/image_proc" />
        <remap from="out/camera_info" to="/camera/camera_info" />
        <param name="binning_factor_x" value="4" />
        <param name="binning_factor_y" value="4" />
    </node>
</launch>
